Broken httpx tests locally

### Describe your environment

OS: (e.g, Ubuntu)
Python version: (e.g., Python 3.8.10) 3.11
Package version: (e.g., 0.46.0) latest main


### What happened?

httpx tests does not pass

### Steps to Reproduce

tox -e py311-test-instrumentation-httpx-1

### Expected Result

Green tests

### Actual Result

```
_________________ TestSyncInstrumentationIntegration.test_instrument_client_with_proxy _________________

self = <tests.test_httpx_integration.TestSyncInstrumentationIntegration testMethod=test_instrument_client_with_proxy>

    def test_instrument_client_with_proxy(self):
        proxy_mounts = self.create_proxy_mounts()
        client = self.create_client(mounts=proxy_mounts)
>       self.assert_proxy_mounts(
            client._mounts.values(),
            2,
            (httpx.HTTPTransport, httpx.AsyncHTTPTransport),
        )

instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:978: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:758: in assert_proxy_mounts
    self.assertEqual(len(mounts), num_mounts)
E   AssertionError: 5 != 2
_______________________ TestSyncInstrumentationIntegration.test_instrument_proxy _______________________

self = <tests.test_httpx_integration.TestSyncInstrumentationIntegration testMethod=test_instrument_proxy>

    def test_instrument_proxy(self):
        proxy_mounts = self.create_proxy_mounts()
        HTTPXClientInstrumentor().instrument()
        client = self.create_client(mounts=proxy_mounts)
        self.perform_request(self.URL, client=client)
        self.assert_span(num_spans=1)
>       self.assert_proxy_mounts(
            client._mounts.values(),
            2,
            (SyncOpenTelemetryTransport, AsyncOpenTelemetryTransport),
        )

instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:968: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:758: in assert_proxy_mounts
    self.assertEqual(len(mounts), num_mounts)
E   AssertionError: 5 != 2
________________ TestSyncInstrumentationIntegration.test_uninstrument_client_with_proxy ________________

self = <tests.test_httpx_integration.TestSyncInstrumentationIntegration testMethod=test_uninstrument_client_with_proxy>

    def test_uninstrument_client_with_proxy(self):
        proxy_mounts = self.create_proxy_mounts()
        HTTPXClientInstrumentor().instrument()
        client = self.create_client(mounts=proxy_mounts)
>       self.assert_proxy_mounts(
            client._mounts.values(),
            2,
            (SyncOpenTelemetryTransport, AsyncOpenTelemetryTransport),
        )

instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:998: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:758: in assert_proxy_mounts
    self.assertEqual(len(mounts), num_mounts)
E   AssertionError: 5 != 2
________________ TestAsyncInstrumentationIntegration.test_instrument_client_with_proxy _________________

self = <tests.test_httpx_integration.TestAsyncInstrumentationIntegration testMethod=test_instrument_client_with_proxy>

    def test_instrument_client_with_proxy(self):
        proxy_mounts = self.create_proxy_mounts()
        client = self.create_client(mounts=proxy_mounts)
>       self.assert_proxy_mounts(
            client._mounts.values(),
            2,
            (httpx.HTTPTransport, httpx.AsyncHTTPTransport),
        )

instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:978: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:758: in assert_proxy_mounts
    self.assertEqual(len(mounts), num_mounts)
E   AssertionError: 5 != 2
______________________ TestAsyncInstrumentationIntegration.test_instrument_proxy _______________________

self = <tests.test_httpx_integration.TestAsyncInstrumentationIntegration testMethod=test_instrument_proxy>

    def test_instrument_proxy(self):
        proxy_mounts = self.create_proxy_mounts()
        HTTPXClientInstrumentor().instrument()
        client = self.create_client(mounts=proxy_mounts)
        self.perform_request(self.URL, client=client)
        self.assert_span(num_spans=1)
>       self.assert_proxy_mounts(
            client._mounts.values(),
            2,
            (SyncOpenTelemetryTransport, AsyncOpenTelemetryTransport),
        )

instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:968: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:758: in assert_proxy_mounts
    self.assertEqual(len(mounts), num_mounts)
E   AssertionError: 5 != 2
_______________ TestAsyncInstrumentationIntegration.test_uninstrument_client_with_proxy ________________

self = <tests.test_httpx_integration.TestAsyncInstrumentationIntegration testMethod=test_uninstrument_client_with_proxy>

    def test_uninstrument_client_with_proxy(self):
        proxy_mounts = self.create_proxy_mounts()
        HTTPXClientInstrumentor().instrument()
        client = self.create_client(mounts=proxy_mounts)
>       self.assert_proxy_mounts(
            client._mounts.values(),
            2,
            (SyncOpenTelemetryTransport, AsyncOpenTelemetryTransport),
        )

instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:998: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py:758: in assert_proxy_mounts
    self.assertEqual(len(mounts), num_mounts)
E   AssertionError: 5 != 2
```

### Additional context

_No response_

### Would you like to implement a fix?

Yes

--- a/instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py
+++ b/instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py
@@ -723,6 +723,7 @@ class BaseTestCases:
                 spans[1].attributes[SpanAttributes.HTTP_URL], https_url
             )
 
+    @mock.patch.dict("os.environ", {"NO_PROXY": ""}, clear=True)
     class BaseInstrumentorTest(BaseTest, metaclass=abc.ABCMeta):
         @abc.abstractmethod
         def create_client(

